src/t8_cmesh/t8_cmesh_vertex_conn_vertex_to_tree.cxx

```diff
@@ -0,0 +1,225 @@
+  } catch (const std::out_of_range& e) {
+    t8_errorf ("ERROR: Could not find vertex %li for cmesh.\n", global_vertex_id);
+    SC_ABORTF ("Caught exception 'out of range': %s\n", e.what ());
+  }
+}
+
+inline const int
+t8_cmesh_vertex_conn_vertex_to_tree::is_committed () const
+{
+  return state == COMMITTED;
+}
+
+inline bool
+t8_cmesh_vertex_conn_vertex_to_tree::operator== (const t8_cmesh_vertex_conn_vertex_to_tree& other) const
+{
+  return is_equal (other);
+}
+
+inline int
+t8_cmesh_vertex_conn_vertex_to_tree::is_equal (const t8_cmesh_vertex_conn_vertex_to_tree& other) const
+{
+  /* Two instances are equal if and only if their
+   * states are equal and the stored vertices are equal. */
+  return state == other.state && vertex_to_tree == other.vertex_to_tree;
+}
+
+/* Mark as ready for commit. Meaning that all 
+  * global vertex ids have been added.
+  * After commit, no vertex ids can be added anymore. */
+void
+t8_cmesh_vertex_conn_vertex_to_tree::commit (const t8_cmesh_t cmesh)
+{
+  sort_list_by_tree_id ();
+  state = COMMITTED;
+
+  T8_ASSERT (contains_all_vertices (cmesh));
+}
+
+void
+t8_cmesh_vertex_conn_vertex_to_tree::add_vertex_to_tree (const t8_cmesh_t cmesh, t8_gloidx_t global_vertex_id,
+                                                         t8_locidx_t ltreeid, int tree_vertex)
+{
+  T8_ASSERT (!is_committed ());
+  T8_ASSERT (0 <= global_vertex_id);
+  T8_ASSERT (t8_cmesh_treeid_is_local_tree (cmesh, ltreeid) || t8_cmesh_treeid_is_ghost (cmesh, ltreeid));
+  T8_ASSERT (t8_cmesh_is_committed (cmesh));
+
+#if T8_ENABLE_DEBUG
+  t8_eclass_t tree_class = t8_cmesh_get_tree_class (cmesh, ltreeid);
+  int num_tree_vertices = t8_eclass_num_vertices[tree_class];
+  T8_ASSERT (0 <= tree_vertex && tree_vertex < num_tree_vertices);
+#endif
+  if (state != INITIALIZED) {
+    SC_ABORTF ("%s:%i: Trying to add vertex to committed vertex to tree structure.\n", __FILE__, __LINE__);
+  }
+
+  tree_vertex_pair pair (ltreeid, tree_vertex);
+  tree_vertex_list& list_of_globalid = vertex_to_tree[global_vertex_id];
+
+  list_of_globalid.push_back (pair);
+}
+
+/* Compare two tree vertex pairs. 
+ * (tree_id_A, vertex_id_A) is consider smaller than
+ * (tree_id_B, vertex_id_B) if either
+ *  tree_id_A < tree_id_B or
+ *  tree_id_A == tree_id_B and vertex_id_A < vertex_id_B */
+static inline bool
+t8_cmesh_tree_vertex_pair_compare (t8_cmesh_vertex_conn_vertex_to_tree::tree_vertex_pair const& pair_a,
+                                   t8_cmesh_vertex_conn_vertex_to_tree::tree_vertex_pair const& pair_b)
+{
+  return pair_a.first == pair_b.first ?                              /* if tree_id_A == tree_id_B  */
+           pair_a.second < pair_b.second                             /* then check vertex_id_A < vertex_id_B */
+                                      : pair_a.first < pair_b.first; /* else check tree_id_A < tree_id_B */
+}
+
+inline void
+t8_cmesh_vertex_conn_vertex_to_tree::sort_list_by_tree_id ()
```

Comment:
I decided against inlining this function since

a) it is a runtime intensive function using std::sort. So no expected gain from inlining
b) Inlining would require to move it to the header and with it "t8_cmesh_tree_vertex_pair_compare". It does not feel right to do so.